Compute maximum of QPs and nodes

dschwen opened the pull request:
Modifies the MaxQpThread object to max on node count per element as well. This slightly changes the meaning of `_max_qps` (which is now technically `_max_qps_and_nodes`), but we've been using this variable as such already.

Closes #11201

dschwen committed:
Compute maximum of QPs and nodes (#11201)

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/174115/) on 9e8251d wanted to post the following:

View the site [here](http://mooseframework.inl.gov/docs/PRs/11202/site)

This comment will be updated on new commits.

andrsd committed:
Adding a test

The test uses ZeroIC to access _zero variable. Also the q-rule is
on purpose dialed to have less q-points than nodes, because in
InitialCondition for nodal variables we loop over nodes and assign node
index into _qp. For this case, we have to make sure we are not accessing
the _zero out-of-bounds.

andrsd commented:
@dschwen I made a test for this, see 12d830687c8cfab40c3631e22437afcb66f02916.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/174239/) on e656b72 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](http://mooseframework.inl.gov/docs/PRs/11202/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s http://mooseframework.inl.gov/docs/PRs/11202/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format fb6b50d4832c72fd19fc2e8dd6f07b70735a0354`

andrsd commented:
I need to test my tests better, fixed here: eeb832d45927ac812f7ac23149d9cd66867b3e67
